src/main/java/au/org/ala/biocache/service/AuthService.java

```diff
@@ -184,26 +193,111 @@ public void run() {
         } else {
             thread.run();
         }
-
     }
 
-    public List<String> getUserRoles(String userId) {
-        List<String> roles = new ArrayList<>();
-        if(StringUtils.isNotBlank(userDetailsUrl)) {
+    @Deprecated
+    public Set<String> getUserRoles(String userId) {
+        Set<String> roles = new HashSet<>();
+        if (StringUtils.isNotBlank(userDetailsUrl)) {
             final String jsonUri = userDetailsUrl + userDetailsPath + "?userName=" + userId;
             logger.info("authCache requesting: " + jsonUri);
-            roles = (List) restTemplate.postForObject(jsonUri, null, Map.class).get("roles");
+            roles.addAll((List) restTemplate.postForObject(jsonUri, null, Map.class).getOrDefault("roles", Collections.EMPTY_LIST));
         }
         return roles;
     }
 
+    @Deprecated
     public Map<String,?> getUserDetails(String userId) {
         Map<String,?> userDetails = new HashMap<>();
-        if(StringUtils.isNotBlank(userDetailsUrl)){
+        if (StringUtils.isNotBlank(userDetailsUrl)){
             final String jsonUri = userDetailsUrl + userDetailsPath + "?userName=" + userId;
             logger.info("authCache requesting: " + jsonUri);
             userDetails = (Map) restTemplate.postForObject(jsonUri, null, Map.class);
         }
         return userDetails;
     }
+
+    /**
+     * Authentication for download users has 3 routes:
+     *
+     * 1) Check for JWT / OAuth- user is retrieved from UserPrincipal along with a set of roles, supplied email address is ignored...
+     * 2) Legacy API Key and X-Auth-Id - email address retrieved from CAS/Userdetails - email address is ignored...
+     * 3) Email address supplied (Galah) - email address is verified - no sensitive access
+     */
+    public Optional<AlaUser> getDownloadUser(DownloadRequestDTO downloadRequestDTO, HttpServletRequest request) {
```

Comment:
JWT and Legacy API key based authentication does not work here as the `request.getUserPrincipal() instanceof PreAuthenticatedAuthenticationToken` always resolves `false`.  This is the same issue mentioned here: https://github.com/AtlasOfLivingAustralia/biocache-service/pull/754/#discussion_r973878545.
This prevents the download operation from completing successfully.

Reply:
Currently, getDownloadUser() does not resolve successfully all three authentication options are failing.